vignettes/tetrapod-biodiversity.Rmd

```diff
@@ -0,0 +1,384 @@
+
+```{r}
+
+plot(alpha_data$interval_mid_ma, alpha_data$n_taxa, # add the points
+     xlim = rev(range(alpha_data$interval_mid_ma, na.rm = TRUE)), # reverse the x-axis
+     xlab="Time (Ma)", ylab="No. taxa", # axes labels
+     pch = 19, col = "#0e826f") # point style and colour
+
+```
+
+that looks good, it's a bit difficult to figure out what stages contain which data points. `axis_geo()` adds a geological timescale between the plot and the axis. 
+
+Taking our plot from above, let's use this function to add a timescale to the x-axis:
+
+```{r}
+# set margins
+par(mar = c(7.1, 4.1, 4.1, 2.1)) # expand at bottom if adding more interval categories
+
+# alpha diversity plot
+plot(alpha_data$interval_mid_ma, alpha_data$n_taxa, # add the points
+     axes = FALSE,
+     xlim = rev(range(alpha_data$interval_mid_ma, na.rm = TRUE)), # reverse the x-axis
+     xlab = " ", ylab = "No. taxa", # axes labels
+     pch = 19, col = "#0e826f") # point style and colour
+box()
+axis(2) # add a normal axis to the elft side
+axis_geo(side = 1, intervals = list("epochs", "periods"))
+title(xlab = "Time (Ma)", line = 4)
+
+```
+
+Amazing! Now we can clearly see that alpha diversity appears to be a lot higher in the Permian than the Carboniferous. But how has the spatial distribution of tetrapods changed across this boundary? Let's use some of the other `palaeoverse` functions to find out!
+
+# Exploring range size/spatial issues through time
+
+Another factor that we can examine is how range size of taxa changed across the C/P boundary. We can examine this spatial distribution through time using the `tax_range_space` function, which provides several different methods for estimating the distribution of taxonomic occurrences. For now we'll use the method `con` which calculates the geographic range of taxa using a convex hull, but the rest of the methods are described in detail in the function documentation which can be accessed via `?tax_range_space`. 
+
+We can use this in combination with `group_apply` to calculate the average for each time interval for our dataset:
+
+```{r}
+# First, remove any occurrences without a genus
+space_tetrapods <- subset(maj_tetrapods, !is.na(genus))
+
+# Find temporal range of all genera
+space_tetrapods <- group_apply(occdf = space_tetrapods,
+                               group = c("bin_midpoint"),
+                               fun = tax_range_space,
+                               name = "genus",
+                               lng = "p_lng",
+                               lat = "p_lat",
+                               method = "con")
+
+# Have a look at the dataset
+head(space_tetrapods)
+
+```
+As you can see, the combination of `group_apply` and `tax_range_space` produces a dataset with the area of geographic range for each taxonomic unit in kilometers squared for each time interval. From this information we can now plot how the average range size of tetrapods changed throughout the Carboniferous and Permian:
```

Comment:
Maybe worth mentioning that an area of 0 here indicates that there is only one occurrence of that taxon?